internal/testutil/datagen/datagen.sh

```diff
@@ -0,0 +1,211 @@
+
+        echo "Installing KSM"
+        helm dependency update ./deployments/ksm > /dev/null
+        helm upgrade --install ksm ./deployments/ksm -n ksm --create-namespace --wait
+
+        echo "Deploying scraper pods"
+        kubectl apply -f ./deployments/scraper.yaml
+    fi
+
+    echo "Waiting for scraper pod to be ready"
+    kubectl -n scraper wait --for=condition=Ready pod -l "$scrapper_selector"
+    pod=$(scraper_pod "$scrapper_selector")
+    if [[ -z "pod" ]]; then
+        echo "Could not find scraper pod (-l $scrapper_selector)"
+        exit 1
+    fi
+    echo "Found scraper pod $pod"
+
+    echo "Copying datagen.sh to scraper pods"
+    kubectl cp datagen.sh scraper/$pod:/bin/
+}
+
+# cleanup uninstalls the dummy resources and the scraper pod.
+function cleanup() {
+    echo "Removing e2e-resources chart"
+    helm uninstall e2e -n mock || true
+    echo "Removing ksm"
+    helm uninstall ksm -n ksm || true
+    echo "Removing scraper pods"
+    kubectl delete -f ./deployments/scraper.yaml --wait || true
+}
+
+# scrape will curl the specified component and output the response body to standard output.
+function scrape() {
+    bearer="Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)"
+    endpoint=unsupported://
+
+    case $1 in
+    ksm)
+      endpoint=http://ksm-kube-state-metrics.ksm.svc:8080/metrics
+    ;;
+    kubelet)
+      endpoint=https://localhost:10250/pods
+    ;;
+    controlplane)
+      case $2 in
+        etcd)
+          endpoint=http://localhost:2381/metrics
+          ;;
+        apiserver)
+          endpoint=https://localhost:8443/metrics
+          ;;
+        controllermanager)
+          endpoint=https://localhost:10257/metrics
+          ;;
+        scheduler)
+          endpoint=https://localhost:10259/metrics
+        ;;
+          *)
+          echo "Unsupported controlplane component $2" >&2
+          return 1
+          ;;
+      esac
+      ;;
+    *)
+      echo "Unknown scrapable $1" >&2
+      return 1
+      ;;
+    esac
+
+    curl -ksSL "$endpoint" -H "$bearer" $curlargs
+    return $?
+}
+
+# scraper_pod is a helper function that returns the name of a pod matching the scraper label.
+function scraper_pod() {
+    kubectl get pods -l $1 -n scraper -o jsonpath='{.items[0].metadata.name}'
+}
+
+# kubedump is a helper function that dumps the specified Kubernetes resource to a yaml file of the same name.
+function kubedump() {
+    echo "Extracting kubernetes $1"
+    kubectl get "$1" -o yaml --all-namespaces > "$1".yaml
+}
+
+# Testinfo is a helper function that returns a markdown-formatted string with basic info of the environment where this
+# script was run.
+function testinfo() {
+cat <<EOF
+# Static test data generated by $(whoami) on $(date -I)
+
+### \`nri-kubernetes\` commit
+\`\`\`
+$(git rev-parse HEAD)
+\`\`\`
+
+\`git status --short\`
+
+\`\`\`
+$(git status --short)
+\`\`\`
+
+### \`$ kubectl version\`
+\`\`\`
```

Comment:
I do not see any obvious issue with the code, but do not trust my bash opinion 😢